From 7d2a0df24227337cf42b024c91708f542ca4ff90 Mon Sep 17 00:00:00 2001
From: Takashi Iwai <tiwai@suse.de>
Date: Tue, 19 Oct 2021 08:05:35 +0200
Subject: [PATCH] ALSA: memalloc: Drop superfluous snd_dma_buffer_sync() declaration
Git-commit: 7d2a0df24227337cf42b024c91708f542ca4ff90
Patch-mainline: v5.16-rc1
References: bsc#1192354

snd_dma_buffer_sync() is declared twice, and the one outside the ifdef
CONFIG_HAS_DMA could lead to a build error when CONFIG_HAS_DMA=n.
As it's an overlooked leftover after rebase, drop this line.

Fixes: a25684a95646 ("ALSA: memalloc: Support for non-contiguous page allocation")
Reported-by: Stephen Rothwell <sfr@canb.auug.org.au>
Link: https://lore.kernel.org/r/20211019165402.4fa82c38@canb.auug.org.au
Link: https://lore.kernel.org/r/20211019060536.26089-1-tiwai@suse.de
Signed-off-by: Takashi Iwai <tiwai@suse.de>

---
 include/sound/memalloc.h | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/include/sound/memalloc.h b/include/sound/memalloc.h
index df615052dad4..653dfffb3ac8 100644
--- a/include/sound/memalloc.h
+++ b/include/sound/memalloc.h
@@ -97,9 +97,6 @@ static inline void snd_dma_buffer_sync(struct snd_dma_buffer *dmab,
 				       enum snd_dma_sync_mode mode) {}
 #endif
 
-void snd_dma_buffer_sync(struct snd_dma_buffer *dmab,
-			 enum snd_dma_sync_mode mode);
-
 dma_addr_t snd_sgbuf_get_addr(struct snd_dma_buffer *dmab, size_t offset);
 struct page *snd_sgbuf_get_page(struct snd_dma_buffer *dmab, size_t offset);
 unsigned int snd_sgbuf_get_chunk_size(struct snd_dma_buffer *dmab,
-- 
2.26.2

